File: Search Algorithms/recursion/recursionExamples.js
//this mimics the effect of using Math.pow()
function power(base, exponent) {
  if (exponent === 0) return 1;
  return base * power(base, exponent - 1);
}

//this mimics a factorial
function factorial(x) {
  if (x < 0) return 0;
  if (x <= 1) return 1;
  return x * factorial(x - 1);
}

//Product of Array
function productOfArray(arr) {
  if (arr.length === 0) {
    return 1;
  }
  return arr[0] * productOfArray(arr.slice(1));
}

function productArray(arr) {
  if (arr.length === 0) return 1;

  return;
}

File: Search Algorithms/recursion/RecursionExamples.cs
using System;


public class RecursionExamples {


int Power(int baseNum, int exponent) {
  if (exponent == 0) return 1;
  return baseNum * Power(baseNum, exponent - 1);
}

int Factorial(int x) {
  if (x < 0) return 0;
  if (x <= 1) return 1;
  return x * Factorial(x - 1);
}


   int ProductOfList(List<int> nums)
    {
        if (nums.Count== 0)
        {
            return 1;
        }
        nums.RemoveAt(1);
        return nums[0] * ProductOfList(nums);
    }

}